### ISSUE TITLE
`detect_variant` doesn't identify `py -m yt_dlp` as `source`

### ISSUE THREAD

**Issue by Bricio** [yt-dlp/yt-dlp#3994]
### Checklist

- [X] I'm requesting a feature unrelated to a specific site
- [X] I've looked through the [README](https://github.com/yt-dlp/yt-dlp#readme)
- [X] I've verified that I'm running yt-dlp version **2022.05.18** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Description

If we get yt-dlp from git source code, when we check the version with -v parameter it should return the actual commit id, not the release commit id.

We can still have the release version from release date.

This will help to ensure that the log provided is using base branch, or something like that.

### Verbose log

Release commit id:
```shell
D:\ytdlp\yt-dlp>py -m yt_dlp -v
[debug] Command-line config: ['-v']
[debug] User config "C:\Users\User\AppData\Roaming\yt-dlp\config.txt": ['--ffmpeg-location', 'D:\\ytdlp\\ffmpeg\\bin']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] Checking exe version: "D:\ytdlp\ffmpeg\bin\ffmpeg" -bsfs
[debug] Checking exe version: "D:\ytdlp\ffmpeg\bin\ffprobe" -bsfs
[debug] exe versions: ffmpeg N-105453-gcc5eb2e662-20220204 (setts), ffprobe N-105453-gcc5eb2e662-20220204
[debug] Optional libraries: Cryptodome-3.14.1, certifi-2021.10.08, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
```

Actual commit id, after changing the code:
```shell
D:\ytdlp\yt-dlp>py -m yt_dlp -v
[debug] Command-line config: ['-v']
[debug] User config "C:\Users\User\AppData\Roaming\yt-dlp\config.txt": ['--ffmpeg-location', 'D:\\ytdlp\\ffmpeg\\bin']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out utf-8 (No VT), error utf-8 (No VT), screen utf-8 (No VT)
[debug] yt-dlp version 2022.05.18 [b8d9c4cdc]
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] Checking exe version: "D:\ytdlp\ffmpeg\bin\ffmpeg" -bsfs
[debug] Checking exe version: "D:\ytdlp\ffmpeg\bin\ffprobe" -bsfs
[debug] exe versions: ffmpeg N-105453-gcc5eb2e662-20220204 (setts), ffprobe N-105453-gcc5eb2e662-20220204
[debug] Optional libraries: Cryptodome-3.14.1, certifi-2021.10.08, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
```

**Comment by pukkandan** [comment]
Your yt-dlp is NOT running from source. If it is, git HEAD is shown

**Comment by pukkandan** [comment]
```
❯ yt-dlp -v
[debug] Command-line config: ['--ignore-config', '-v']
[debug] Encodings: locale cp65001, fs utf-8, pref cp65001, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version 2022.05.18 [b14d52355] (source)
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Git HEAD: 5ec1b6b71
[debug] Python version 3.10.4 (CPython 64bit) - Windows-10-10.0.22000-SP0
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] exe versions: ffmpeg N-106550-g072101bd52-20220410 (fdk,setts), ffprobe N-106624-g391ce570c8-20220415, phantomjs 2.1.1
[debug] Optional libraries: Cryptodome-3.14.1, brotli-1.0.9, certifi-2021.10.08, mutagen-1.45.1, sqlite3-2.6.0, websockets-10.1
[debug] Proxy map: {}

Usage: yt-dlp [OPTIONS] URL [URL...]

yt-dlp: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.
```

See the line:
> [debug] Git HEAD: 5ec1b6b71

**Comment by Bricio** [comment]
@pukkandan 
> Your yt-dlp is NOT running from source. If it is, git HEAD is shown

I am running from master source code and there is no HEAD line.
![image](https://user-images.githubusercontent.com/216170/172445913-fa14bc4c-5558-4594-9ee1-4f810321c1f8.png)


```
User@Comp MINGW64 /d/ytdlp/yt-dlp (master)
$git remote show upstream
* remote upstream
  Fetch URL: https://github.com/yt-dlp/yt-dlp
  Push  URL: https://github.com/yt-dlp/yt-dlp
  HEAD branch: master
  Remote branches:
    gh-pages tracked
    master   tracked
    release  tracked
  Local ref configured for 'git push':
    master pushes to master (up to date)
User@Comp MINGW64 /d/ytdlp/yt-dlp (master)
$git fetch upstream
User@Comp MINGW64 /d/ytdlp/yt-dlp (master)
$git reset --hard upstream/master
HEAD is now at 0d6bafbfa [expressen] Fix extractor (#4006)
User@Comp MINGW64 /d/ytdlp/yt-dlp (master)
$py -m yt_dlp -v --no-config
[debug] Command-line config: ['-v', '--no-config']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out cp1252 (No VT), error cp1252 (No VT), screen cp1252 (No VT)
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: avconv -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: avprobe -bsfs
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.14.1, certifi-2021.10.08, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}
WARNING: Long argument string detected. Use -- to separate parameters and URLs, like this:
yt-dlp -v -- --no-config

Usage: __main__.py [OPTIONS] URL [URL...]

__main__.py: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.
User@Comp MINGW64 /d/ytdlp/yt-dlp (master)
$^C
User@Comp MINGW64 /d/ytdlp/yt-dlp (master)
$py -m yt_dlp -v --ignore-config
[debug] Command-line config: ['-v', '--ignore-config']
[debug] Encodings: locale cp1252, fs utf-8, pref cp1252, out cp1252 (No VT), error cp1252 (No VT), screen cp1252 (No VT)
[debug] yt-dlp version 2022.05.18 [b14d52355]
[debug] Lazy loading extractors is disabled
[debug] Plugins: ['SamplePluginIE', 'SamplePluginPP']
[debug] Python version 3.8.10 (CPython 64bit) - Windows-7-6.1.7601-SP1
[debug] Checking exe version: ffmpeg -bsfs
[debug] Checking exe version: avconv -bsfs
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: avprobe -bsfs
[debug] exe versions: none
[debug] Optional libraries: Cryptodome-3.14.1, certifi-2021.10.08, sqlite3-2.6.0, websockets-10.3
[debug] Proxy map: {}

Usage: __main__.py [OPTIONS] URL [URL...]

__main__.py: error: You must provide at least one URL.
Type yt-dlp --help to see a list of all options.
User@Comp MINGW64 /d/ytdlp/yt-dlp (master)

```

I just followed the developer instructions https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions

> To run yt-dlp as a developer, you don't need to build anything either. Simply execute
>     python -m yt_dlp

**Comment by dirkf** [comment]
FWIW yt-dl shows the `Git HEAD` line in both (equivalent) cases.
